### ISSUE TITLE
Bug in Plotter for content type 'uncertainty'

### ISSUE THREAD

**Issue by vikusc** [eos/eos#585]
Trying to plot a simple graph of a function f(x) with the content type 'uncertainty' leads to a bug when only having one sample per x-value since the evaluation of the mean and uncertainties expects an object that has a len().

A possible workaround is to wrap the samples into another array.

```
plot_args = {
    'plot': {
        'x': {'label': r'$x$', 'range': [0.0, 1.0] },
        'y': { 'label': r'$f(x)$', 'range': [0.0,  1.0] },
    },
    'contents': [
        {
            'type': 'uncertainty',
            'color': 'C0,
            'band': ['median'],
            'range': [0.0, 1.0],
            'data': {'samples': np.array([samples]), 'xvalues': xvalues}
        }
    ]
}

eos.plot.Plotter(plot_args).plot()```

**Comment by dvandyk** [comment]
The proper way of doing this is probably to create a new plot type ``function-samples``? Name to be work shopped...

**Comment by mreboud** [comment]
Since this is basically a curve that we plot, can we name it `curve`? For the moment we do have simple names as `point`, `band`, `line`, etc...

**Comment by vikusc** [comment]
Or `graph`? One plots (x,f(x)) for each point in the x range and thats formally the graph of the function f?

**Comment by mreboud** [comment]
Once the issue is fixed, I can also add commits d9e7b60135e6b79e37eafd05eb889fdab5e6fa66 and e25b2d3327f92231ab0c3db11eeb86ff5b606806 to the PR.
